pennylane/tape/tapes/rewind.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Quantum tape for implementing the gradient method outlined in https://arxiv.org/abs/2009.02823,
+referred to here as the "rewind" method.
+"""
+import numpy as np
+
+# pylint: disable=protected-access
+import pennylane as qml
+
+from .jacobian_tape import JacobianTape
+
+
+class RewindTape(JacobianTape):
+    r"""Quantum tape for computing gradients using the `rewind <https://arxiv.org/abs/2009.02823>`__
+    method.
+
+    After a forward pass, the circuit is "rewound" by iteratively applying the inverse gate to scan
+    backwards through the circuit. This method is similar to the reversible method, but has a lower
+    time overhead and a similar memory overhead.
+
+    .. note::
+
+        The rewind analytic differentation method has the following restrictions:
+
+        * As it requires knowledge of the statevector, only statevector simulator devices can be
+          used.
+
+        * Only expectation values are supported as measurements.
+
+    This class extends the :class:`~.jacobian` method of the quantum tape to support analytic
+    gradients of qubit operations using the rewind method of analytic differentiation.
+    This gradient method returns *exact* gradients, however requires use of a statevector simulator.
+    Simply create the tape, and then call the Jacobian method:
+
```

Comment:
Should we show how to create a tape as a rewind tape?

```suggestion
>>> dev = qml.Device('default.qubit', wires=[0]) 
>>> with qml.RewindTape() as tape:
          qml.RX(0.1, wires=0)
```